apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -2974,6 +2976,73 @@ sophia_aens(_Cfg) ->
     {error, <<"out_of_gas">>} = ?call(call_contract, Acc, Ct, revoke, {tuple, []}, {Ct, NHash, 0}, #{ height => 13 }),
     ok.
 
+sophia_state_handling(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc      = ?call(new_account, 1000000),
+    Ct0      = ?call(create_contract, Acc, remote_state, {}, #{ amount => 100000 }),
+    %% Test an init function that calls a remote contract to compute the state
+    Ct1      = ?call(create_contract, Acc, state_handling, {Ct0, 1}, #{ amount => 100000 }),
+    Ct2      = ?call(create_contract, Acc, state_handling, {Ct0, 2}, #{ amount => 100000 }),
+    MapT     = {map, word, word},
+    StateT   = {tuple, [word, string, MapT]},
+    UnitT    = {tuple, []},
+
+    %% Test that we can read the state
+    ?assertEqual({1, <<"undefined">>, #{}}, ?call(call_contract, Acc, Ct1, read, StateT, {})),
+    ?assertEqual(1,                         ?call(call_contract, Acc, Ct1, read_i, word, {})),
+    ?assertEqual(<<"undefined">>,           ?call(call_contract, Acc, Ct1, read_s, string, {})),
+    ?assertEqual(#{},                       ?call(call_contract, Acc, Ct1, read_m, MapT, {})),
+    ?assertEqual({2, <<"undefined">>, #{}}, ?call(call_contract, Acc, Ct2, read, StateT, {})),
+    ?assertEqual(2,                         ?call(call_contract, Acc, Ct2, read_i, word, {})),
+    ?assertEqual(<<"undefined">>,           ?call(call_contract, Acc, Ct2, read_s, string, {})),
+    ?assertEqual(#{},                       ?call(call_contract, Acc, Ct2, read_m, MapT, {})),
+
+    %% Test that we can update the state
+    ?call(call_contract, Acc, Ct1, update_i, UnitT, {7}),
+    ?call(call_contract, Acc, Ct1, update_s, UnitT, {<<"defined">>}),
+    ?call(call_contract, Acc, Ct1, update_m, UnitT, {#{3 => 4, 1 => 2}}),
+    ?assertMatch({7, <<"defined">>, #{3 := 4, 1 := 2}},
```

Comment:
Why match and not equal? Also below. This may not check if any map keys added.

```
$ erl
...
2> #{1 := 2} = #{2 => 3, 1 => 2}.
#{1 => 2,2 => 3}
```

Reply:
No reason, started with Equal, and then my brain changed to Match after the first three or so?!